exodus_lambda/functions/verify_source.py

```diff
@@ -0,0 +1,64 @@
+import os
+import subprocess
+
+import boto3
+
+
+def lambda_handler(event, context):
+    """Verifies the commit delivered to the pipeline by GitHub has a
+    tag signed by a trusted entity.
+
+    Receives the commit ID from the pipeline via UserParameters.
+
+    This function must be manually deployed prior to pipeline creation.
+    """
+    # pylint: disable=unused-argument
+
+    job_id = event["CodePipeline.job"]["id"]
+    job_data = event["CodePipeline.job"]["data"]
+    git_commit = job_data["actionConfiguration"]["configuration"][
+        "UserParameters"
+    ]
+
+    bucket_name = "exodus-pipeline-artifacts"
+    whitelist_name = ".whitelist.gpg"
+
+    s3_client = boto3.client("s3")
+    pipeline_client = boto3.client("codepipeline")
+
+    try:
+        # 1) Download and import whitelisted GPG keys
+        s3_client.download_file(bucket_name, whitelist_name)
+        subprocess.run(
+            ["gpg", "--import", whitelist_name], check=True, text=True
+        )
+
+        # 2) Clone exodus-lambda and checkout deploy branch
+        repo_url = "https://github.com/release-engineering/exodus-lambda.git"
+        subprocess.run(
+            ["git", "clone", repo_url, "-b", "deploy"], check=True, text=True
```

Comment:
Why are we passing text=True - we aren't reading stdout or stderr from the process, so does that do anything here?

Reply:
Nope. I'll remove it.